src/types.rs

```diff
@@ -459,6 +469,9 @@ macro_rules! impl_from_skyhash {
         $(impl FromSkyhashBytes for $ty {
             fn from_element(element: Element) -> SkyResult<$ty> {
                 let ret = match element {
+                    // Note: We're assuming primitive types are internally
+                    // represented as strings, which we then parse.
+                    // TODO: Represent primitives more efficiently.
```

Comment:
This is just for convenience. We can simply do a `to_string` but it's not applicable for other types in this specific shorthand. Second, no, primitive types are **not** represented as strings internally. The formatting infrastructure is used to produce strings for types like integers.